modules/cudaoptflow/src/pyrlk.cpp

```diff
@@ -142,66 +158,70 @@ namespace
         GpuMat temp2 = nextPts.reshape(1);
         cuda::multiply(temp1, Scalar::all(1.0 / (1 << maxLevel_) / 2.0), temp2, 1, -1, stream);
 
+
         ensureSizeIsEnough(1, prevPts.cols, CV_8UC1, status);
         status.setTo(Scalar::all(1), stream);
 
         if (err)
             ensureSizeIsEnough(1, prevPts.cols, CV_32FC1, *err);
 
-        // build the image pyramids.
+        if (prevPyr.size() != maxLevel_ + 1 || nextPyr.size() != maxLevel_ + 1)
```

Comment:
This line generates signed/unsigned comparison warning. Please cast `maxLevel_ + 1` to `size_t` to fix it:

```
if (prevPyr.size() != size_t(maxLevel_ + 1) || nextPyr.size() != size_t(maxLevel_ + 1))
```